functions/executor/call.py

```diff
@@ -104,33 +111,183 @@ def exec_dag_function(pusher_cache, kvs, triggers, function, schedule):
         l = LWWPairLattice(generate_timestamp(0), serialize_val(result))
         kvs.put(schedule.response_id, l)
 
+def _exec_func_normal(kvs, func, args):
+    func_args = ()
 
-def _process_args(arg_list):
-    return list(map(lambda v: get_serializer(v.type).load(v.body), arg_list))
+    to_resolve = []
 
+    # resolve any references to KVS objects
+    key_index_map = {}
+    for i, arg in enumerate(args):
```

Comment:
I'm not sure I understand why this code is written this way. Why do you create a tuple with value, then replace values? Why not just resolve the references in place? I don't see any reason why this code should be different from the [existing code](https://github.com/fluent-project/fluent/blob/master/functions/executor/call.py#L113-L120).

Reply:
Because the `async_cache` supports reading a set of keys, so this way we only issue one request to the cache? Previously we were issuing k requests..In this setting getting some keys but not others won't speed up the function execution so I am just batching keys into one req

Reply:
Ah, got it. Makes sense!